### ISSUE TITLE
NPM Restore 15 times slower on versions > 6.14.13

### ISSUE THREAD

**Issue by andtii** [npm/cli#3939]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running npm restore on versions > 6.14.13 we can see 12-15 times slower restore times.
The issue was found in this issue > https://github.com/actions/virtual-environments/issues/4284

### Expected Behavior

Same restore time as 6.14.13 = below 60 seconds

### Steps To Reproduce

1. Install npm 6.14.13 > npm install npm@6.14.13 -g
2. Run a npm restore on this example repo > https://github.com/preciofishbone/NpmRestoreIsSlow
2. IT takes under 60 seconds
3. Install npm 6.14.15 > npm install npm@6.14.15 -g
4. Run a npm restore on this example repo > https://github.com/preciofishbone/NpmRestoreIsSlow
5. Restore takes 650-900 seconds 
6. Same issue on latest npm and node

### Environment

- OS:  Ubuntu 20.04
- Node:  14.18.1 or any newer
- npm:  6.14.13/6.14.15 or any newer

**Comment by ljharb** [comment]
What about with npm 7 or 8? npm 6 is basically no longer supported.

**Comment by andtii** [comment]
Tried this with Node 17.0.1 and Npm 7.24.2 but still same issue, I will try with 8.1.1 and come back with the result

**Comment by andtii** [comment]
@ljharb I tried this with 8.1.1 and it is still slow

**Comment by lukekarrys** [comment]
I ran `npm i --loglevel silly --timing --force` (the `--force` is because there are some `ERESOLVE` errors in the tree, but that shouldn't be relevant). I did find this line in the log which takes up almost all the time

```
8634 timing reifyNode:node_modules/material-design-icons Completed in 349261ms
...
10451 timing npm Completed in 395395ms
```

I haven't looked into why this is faster on npm 6, but I did see that the unpacked size for `material-design-icons` is `376.3 MB`, so it makes some sense why that is taking a long time. But I will continue to dig in and see why npm 6 is so much faster.

**Comment by lukekarrys** [comment]
I installed `material-design-icons@3.0.1` using `npm i --loglevel silly --timing --offline --no-audit` to isolate the timing to (mostly) extracting the tarball for that package and here's what I found:

```sh
> v6.14.13
30066ms

> v6.14.15
900499ms

> v7.20.1
24748ms

> v7.21.0
910164ms

> v7.21.1
281097ms

> v8.1.2
289085ms
```

This comment https://github.com/npm/cli/issues/3676#issuecomment-946237180 has more info as to why this regression happened, but the tl;dr is that it was due to some security related [`node-tar` fixes](https://github.com/npm/node-tar/security/advisories?state=published) released in `v6.14.15` and `v7.21.0`. We were able to bring back some (but not all) of the performance in `v7.21.1` but packages like `material-design-icons` are affected more prominently due to the number of files they contain.

We no longer support npm 6 outside of security related fixes, so there are no plans to do an npm 6 release with the performance fixes in npm 7/8. My recomendation would be to update to npm 8 and/or to look into alternatives to `material-design-icons` (https://github.com/google/material-design-icons/issues/1129#issuecomment-875085415).
